product.py

```diff
@@ -534,13 +534,32 @@ def findXenSourceBackups():
         b = None
         try:
             b = util.TempMount(p, 'backup-', ['ro'], 'ext3')
-            if os.path.exists(os.path.join(b.mount_point, '.xen-backup-partition')):
-                backup = XenServerBackup(p, b.mount_point)
-                logger.log("Found a backup: %s" % (repr(backup),))
-                if backup.version >= XENSERVER_MIN_VERSION and \
-                        backup.version <= THIS_PLATFORM_VERSION:
-                    backups.append(backup)
-        except:
+            if not os.path.exists(os.path.join(b.mount_point, '.xen-backup-partition')):
+                raise StopIteration()
```

Comment:
I found this raising `StopIteration` and catching it weird and hard to read. Why not using a `try/finally` doing the `b.umount()` in the finally and using `continue` here?